knfc/knfconverter.js

```diff
@@ -0,0 +1,220 @@
+ Node_c.setIDRec = function (id)  {
+    this.id = id;
+    if (this.left !== null) {
+       this.left.setIDRec(id + "0");
+    }
+    if (this.right !== null) {
+       this.right.setIDRec(id + "1");
+    }
+ };
+ Node_c.setID = function (id)  {
+    this.id = id;
+    jv_System_c.out.println("ID SET FOR" + Node_c.toString.call(this));
+ };
+ 
+ Node_c._NodeInit = function() {
+    this.left = null;
+       ;
+    this.right = null;
+       ;
+    this.parent = null;
+       ;
+    this.s = null;
+       ;
+    this.id = null;
+       ;
+ };
+ 
+ // Generated JS from Java: Abfahrt2 -----
+ function Abfahrt2() {
+    jv_Object.call(this);
+ }
+ 
+ var Abfahrt2_c = sc_newClass("Abfahrt2", Abfahrt2, jv_Object, null);
+ 
+ Abfahrt2_c.toKNF = function (ns)  {
+    var ret = new jv_ArrayList();
+    ret.add("{T}");
+    for (var _i = ns.iterator(); _i.hasNext();) {
+       var n = _i.next();
+       var a = n.left.s + n.left.id;
+       var b = n.right.left.s + (n.right.left.id !== null ? n.right.left.id : "");
+       var c = null;
+       if (n.right.right !== null) {
+          c = n.right.right.s + (n.right.right.id !== null ? n.right.right.id : "");
+       }
+       var not = "Â¬";
+       var S = ", ";
+       switch(n.right.s.charAt(0)) {
+          case '!':
+             {
+                ret.add("{" + a + S + b + "}");
+                ret.add("{" + not + a + S + not + b + "}");
+                break;
+          }
+          case '<':
+             {
+                ret.add("{" + not + a + S + not + b + S + c + "}");
+                ret.add("{" + not + a + S + b + S + not + c + "}");
+                ret.add("{" + a + S + b + S + c + "}");
+                ret.add("{" + a + S + not + b + S + not + c + "}");
+                break;
+          }
+          case '-':
+             {
+                ret.add("{" + not + a + S + not + b + S + c + "}");
+                ret.add("{" + a + S + b + "}");
+                ret.add("{" + a + S + not + c + "}");
+                break;
+          }
+          case '%':
+             {
+                ret.add("{" + not + a + S + not + b + S + c + "}");
+                ret.add("{" + not + a + S + not + b + S + not + c + "}");
+                ret.add("{" + a + S + not + b + S + c + "}");
+                ret.add("{" + a + S + b + S + not + c + "}");
+                break;
+          }
+          case '|':
+             {
+                ret.add("{" + not + a + S + b + S + c + "}");
+                ret.add("{" + a + S + not + b + "}");
+                ret.add("{" + a + S + not + c + "}");
+                break;
+          }
+          case '&':
+             {
+                ret.add("{" + a + S + not + b + S + not + c + "}");
+                ret.add("{" + not + a + S + b + "}");
+                ret.add("{" + not + a + S + c + "}");
+                break;
+          }
+       }
+    }
+    return ret;
+ };
+ Abfahrt2_c.generateClauses = function (n)  {
+    var ret = new jv_ArrayList();
```

Comment:
in js kannst du einfach ein Array nehmen.
https://www.w3schools.com/js/js_arrays.asp